internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -708,26 +707,18 @@ func isUsingOldAPISchemaStructure(d *schema.ResourceData) bool {
 
 func resourceUpdateOrUpgrade(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	if upgradeRequest := getUpgradeRequest(d); upgradeRequest != nil {
-		upgradeCtx := context.WithValue(ctx, upgradeRequestCtxKey, upgradeRequest)
-		return resourceUpgrade(upgradeCtx, d, meta)
```

Comment:
small refactor: upgrade request was being passed through the context making code harder to follow.